### ISSUE TITLE
is_online() fails with raise NotImplementedError()

### ISSUE THREAD

**Issue by tuxmaster5000** [python-escpos/python-escpos#622]
I have:
- [X ] searched open and closed issues for duplicates

### Bug description
It looks like the function is missing in the File "driver":

> File "/usr/lib/python3.11/site-packages/escpos/escpos.py", line 177, in _read
>     raise NotImplementedError()
> NotImplementedError

### Steps to reproduce
```
from escpos.escpos import EscposIO
from escpos import printer

with EscposIO(printer.File("<Path>", profile="TM-U220B")) as x:
     while not x.printer.is_online():
     pass
```

### Device info
 **Printer:** Epson Model TM-U220B

 **python-escpos version:** 3.1

 **python version:** 3.11.7
 
 **operating system:** Fedora-38

**Comment by hasansezertasan** [comment]
`escpos.printer.file.File` used for parallel port printer or other printers that are directly attached to the file system.

```python
# /// script
# requires-python = ">=3.12"
# dependencies = ["python-escpos"]
# ///
import tempfile

from escpos.escpos import EscposIO
from escpos.printer import File


def main() -> None:
    """Test the EscposIO class with a File printer."""
    with tempfile.NamedTemporaryFile() as f:
        print(f.name)
        with EscposIO(File(f.name, profile="TM-U220B")) as writer:
            writer.printer.text("Hello from main.py!\n")
            writer.printer.text("Hello from main.py!\n")
            writer.printer.cut()
        f.seek(0)
        print(f.read())


if __name__ == "__main__":
    main()
```

Let me know if I'm incorrect but in the end, it's only a file. So `is_online` only depends on file availability and access permissions.

I think it can return `True` by default for convenience.

**Comment by tuxmaster5000** [comment]
I have tested it with an direct attached device over USB.
This is because the Epson command set recognises the query as to whether the device is online or not. And the model also has the appropriate switch for this.
I think this will be the command:
https://download4.epson.biz/sec_pubs/pos/reference_en/escpos/dle_eot.html

**Comment by belono** [comment]
Hi @tuxmaster5000 and @hasansezertasan !

Take a look at #656. It's a similar case.

Maybe we could find a workaround as I did for the `CupsPrinter` connector.

**Comment by tuxmaster5000** [comment]
@belono Hm, but your link shows an Windows problem. Here we have Linux.

**Comment by belono** [comment]
> [@belono](https://github.com/belono) Hm, but your link shows an Windows problem. Here we have Linux.

Yes, but the source of the problem is quite similar: the `File` connector lacks an implementation of the `_read()` method, just like the `Win32Raw` connector.

> Let me know if I'm incorrect but in the end, it's only a file. So is_online only depends on file availability and access permissions.

Sadly is not as easy.
As @tuxmaster5000 said:

> This is because the Epson command set recognises the query as to whether the device is online or not. And the model also has the appropriate switch for this.

We write a few bytes to the printer through its connector (usb, serial, network...) to query its status, the printer processes the query and then we have to read its data buffer to get the response back.

I've been digging a bit into this and the good news is that the `File` connector can act as a low-level connector in Linux if the user has sufficient access permissions ,so we could implement a `_read()` method that reads the printer's data buffer to expose the `query_status()`, `is_online()` and `paper_status()` methods to the `File` connector.

This test code is working on my old Epson TM-U210 Parallel printer running Raspbian Bullseye with Python 3.9:
```python
import contextlib
import os


# Control characters
EOT: bytes = b"\x04"
DLE: bytes = b"\x10"
# Status Command
RT_STATUS: bytes = DLE + EOT
RT_STATUS_ONLINE: bytes = RT_STATUS + b"\x01"
RT_STATUS_PAPER: bytes = RT_STATUS + b"\x04"


@contextlib.contextmanager
def managed_fd(path, flags):
    fd = os.open(path, flags)
    try:
        yield fd
    finally:
        os.close(fd)

def request_status(fd, status):
    print("Request printer status... ", end = '')
    os.write(fd, status)
    print("Ok")
    os.lseek(fd, 0, 0)  # mandatory rewind
    print("Reading printer status... ", end = '')
    print(os.read(fd, 16).hex())  # read the buffer


with managed_fd("/dev/usb/lp0", os.O_RDWR) as fd:
    request_status(fd, RT_STATUS_ONLINE)
    request_status(fd, RT_STATUS_PAPER)

```
